tests/test_semaphore_tracker.py

```diff
@@ -0,0 +1,209 @@
+"""Tests for the SemaphoreTracker class"""
+import errno
+import gc
+import io
+import os
+import pytest
+import re
+import signal
+import sys
+import time
+import warnings
+import weakref
+
+from loky import ProcessPoolExecutor
+import loky.backend.semaphore_tracker as semaphore_tracker
+from loky.backend.semlock import sem_unlink
+from loky.backend.context import get_context
+
+
+def get_sem_tracker_pid():
+    semaphore_tracker.ensure_running()
+    return semaphore_tracker._semaphore_tracker._pid
+
+
+class TestSemaphoreTracker:
+
+    @pytest.mark.skipif(sys.platform == "win32",
+                        reason="no semaphore_tracker on windows")
+    def test_child_retrieves_semaphore_tracker(self):
+        parent_sem_tracker_pid = get_sem_tracker_pid()
+        executor = ProcessPoolExecutor(max_workers=2)
+        child_sem_tracker_pid = executor.submit(get_sem_tracker_pid).result()
+
+        # First simple pid retrieval check (see #200)
+        assert child_sem_tracker_pid == parent_sem_tracker_pid
+
+        # Register a semaphore in the parent process, and un-register it in the
+        # child process. If the two processes do not share the same
+        # semaphore_tracker, a cache KeyError should be printed in stderr.
+        import subprocess
+        semlock_name = 'loky-mysemaphore'
+        cmd = '''if 1:
+        import os, sys
+
+        from loky import ProcessPoolExecutor
+        from loky.backend import semaphore_tracker
+        from loky.backend.semlock import SemLock
+
+        semaphore_tracker.VERBOSE=True
+        semlock_name = "{}"
+
+        # The benefit of using _SemLock objects in this test is that they do
```

Comment:
`_SemLock` -> `semlock.SemLock`

Reply:
Actually, as you are not `unlinking` the `semlock.SemLock` at the end of the test, you are leaking a semaphore. As you `register`/`unregister` a name, you could simply pass the name and not instantiate the actual object?

Reply:
Good idea.

Reply:
Ok I missed the finally sorry.